analytics-listeners/src/main/java/org/exoplatform/analytics/listener/social/AnalyticsAttachmentListener.java

```diff
@@ -0,0 +1,117 @@
+import org.exoplatform.analytics.utils.AnalyticsUtils;
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.services.listener.Asynchronous;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.social.attachment.AttachmentPlugin;
+import org.exoplatform.social.attachment.AttachmentService;
+import org.exoplatform.social.attachment.model.ObjectAttachmentId;
+import org.exoplatform.social.core.activity.model.ExoSocialActivity;
+import org.exoplatform.social.core.manager.ActivityManager;
+
+import java.util.Date;
+import java.util.Map;
+
+@Asynchronous
+public class AnalyticsAttachmentListener extends Listener<String, ObjectAttachmentId> {
+
+  public static final String      TASK_COMMENT_OBJECT_TYPE     = "taskComment";
+
+  public static final String      ACTIVITY_COMMENT_OBJECT_TYPE = "comment";
+
+  public static final String      TASK_OBJECT_TYPE             = "task";
+
+  public static final String      ACTIVITY_OBJECT_TYPE         = "activity";
+
+  public static final String      RULE_OBJECT_TYPE             = "rule";
+
+  public static final String      STATISTICS_OPERATION         = "attachImages";
+
+  private final ActivityManager   activityManager;
+
+  private final AttachmentService attachmentService;
+
+  public AnalyticsAttachmentListener(ActivityManager activityManager, AttachmentService attachmentService) {
+    this.activityManager = activityManager;
+    this.attachmentService = attachmentService;
+  }
+
+  @Override
+  @ExoTransactional
+  public void onEvent(Event<String, ObjectAttachmentId> event) throws Exception {
+
+    String username = event.getSource();
+    ObjectAttachmentId objectAttachment = event.getData();
+
+    if (objectAttachment == null) {
+      return;
+    }
+
+    String objectModule = getModule(objectAttachment.getObjectType());
+    String objectSubModule = getSubModule(objectAttachment.getObjectType(), objectAttachment.getObjectId());
+
+    Map<String, AttachmentPlugin> attachmentPlugins = attachmentService.getAttachmentPlugins();
+    AttachmentPlugin attachmentPlugin = attachmentPlugins.get(objectAttachment.getObjectType());
+    long spaceId = attachmentPlugin.getSpaceId(objectAttachment.getObjectId());
+
+    StatisticData statisticData = new StatisticData();
+    statisticData.setModule(objectModule);
+    statisticData.setSubModule(objectSubModule);
+    statisticData.setOperation(STATISTICS_OPERATION);
+    statisticData.setTimestamp(new Date().getTime());
+    statisticData.setUserId(AnalyticsUtils.getUserIdentityId(username));
+    statisticData.addParameter("spaceId", spaceId);
+  }
+
+  private String getModule(String objectType) {
+    return switch (objectType) {
+    case TASK_COMMENT_OBJECT_TYPE, TASK_OBJECT_TYPE -> "tasks";
+    case ACTIVITY_OBJECT_TYPE -> "social";
+    case RULE_OBJECT_TYPE -> "gamification";
+    default -> "";
+    };
+  }
```

Comment:
It would be better to use the object type as is in the analytics parameters and to embed the module name `social` and sub module name `attachment`.

```suggestion
```

Reply:
done